content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,519 @@
+```
+![df_sample](df_sample5.png)
+```python
+stats.ttest_1samp(a=df_, popmean=1.199)
+```
+![onesampttest](onsamplettest.png)
+
+**H0:** The mean of petal_width is 1.199
+
+**H1:** The mean of petal_width is not 1.199
+
+Analyzing one-sample t-test, it finds out if the hypothesized mean is the same or different from the group's mean. From the example above, we've selected the petal_width as the population sample to perform the test on. 
+From the test, since the P-value beats the alpha level set at 0.05,
+
+we fail to reject the null hypothesis because we don't have enough evidence to prove otherwise, moreover, if you check, the P-value is very close to 1.0,
+
+This means that the mean of petal_width is most likely 1.199.
+ 
+ The evidence is too strong to be rejected, it is almost accurate. (Obviously, I copied the mean from the summary).
+
+#### Performing a Two-Sample T-Test
+
+We need to first of all check their variances.
+
+The ratio of the higher to the lower variance must be less than 4:1 before we can assume the two samples have the same variance.
+```python
+class1 = df.petal_width
+class2 = df.sepal_length
+```
+find variance for each group
+```python
+print(np.var(class1), np.var(class2))
+```
+![var](variancetwosamplettest.png)
+```python
+ratio_check = 0.6811222222222222 / 0.5785315555555559
+
+ratio_check
+```
+![ratiocheck](ratiocheck.png)
+Obviously, the ratio is less than 4:1, thus the variances are assumed to be equal.
+```python
+stats.ttest_ind(a=class1, b=class2, equal_var=True)
+```
+![twosample](twosamplettest.png)
+
+Evaluating this:
+
+**H0:** The mean of the two samples (petal_width and sepal_length) are equal. 
+
+**H1:** The mean of the two samples (petal_width and sepal_length) are not equal.
+
+Our P-value ( p = 3.7974378831185126e-148) is far less than alpha = 0.05, 
+
+Checking of the variance is only making sure that two populations are probably spread out or normally distributed, this will help prevent bias. Hence, the null hypothesis will be assuming that since the two groups are normally distributed, they likely should have the same mean. However, interpreting any form of t-test largely depends on the p-value, the result above explains that the means of the two samples(petal_width and sepal_length) are different. Thus, we will reject the Null Hypothesis which says the mean of the two samples are the same because their variances are assumed to be equal, there is no evidence sustainable enough to prove the two populations have the same mean.
+H1 (alternate hypothesis) is however true.
+
+This allows us to figure out questions like: 
+"Is there a significant or random by the chance difference in the average time spent on mathematics exam from that of English exam?."
+
+#### ANOVA Test
+
+ANOVA is a word coined from `analysis of variance`. It is a statistical concept that shows the differences between the means of more than two independent classes, using variance analysis on samples from those classes.
```

Comment:
The phrase 'analysis of variance' can be put into single quotes and not inverted quotes. Also, can be reworded as 'ANalysis Of VAriance'

Reply:
done